dlf/plugins/oai/class.tx_dlf_oai.php

```diff
@@ -967,13 +967,9 @@ protected function verbIdentify() {
 
 		} else {
 
-			if (TYPO3_DLOG) {
-
-				t3lib_div::devLog('[tx_dlf_oai->verbIdentify()] Incomplete plugin configuration', $this->extKey, SYSLOG_SEVERITY_ERROR);
-
-			}
+			t3lib_div::devLog('[tx_dlf_oai->verbIdentify()] Incomplete plugin configuration', $this->extKey, SYSLOG_SEVERITY_ERROR);
```

Comment:
TYPO3_DLOG was introduced in 2004 to implement logging "with a minimum loss of speed if no logging is asked for". In our case here speed is certainly not an argument.

The new code will write a log message only if the devlog extension is enabled and if there is a bad configuration, so normal production installations won't get logging here even without TYPO3_DLOG.

And those people who install devlog might be interested to see error messages in their TYPO3 log, but usually they don't want debug messages from all parts of TYPO3 which they would get when TYPO3_DLOG is set.

That's why I think removing the check for TYPO3_DLOG here is good. Otherwise only the end user but not the system administrator would see that there is a problem.

In addition, I looked at other TYPO3 code to see how logging is handled. The extension t3jquery writes error logging without checks. The extension realurl has a wrapper function and uses its own configuration to enable logging. The TYPO3 core uses several strategies for logging, some check TYPO3_DLOG or similar settings, some use wrapper functions, others don't check anything.

Maybe a helper function would be a good solution for dlf, too. Then we could remove a lot of if statements and do a more advanced check only in the wrapper.